README.md

```diff
@@ -1,88 +1,158 @@
-Node.js client for [FaunaDB](https://faunadb.com).
+# FaunaDB Javascript Driver
 
-View documentation [here](https://faunadb.github.io/faunadb-js/).
+A Javascript driver for [FaunaDB](https://faunadb.com).
 
-See an example [here](https://github.com/faunadb/faunadb-js/blob/master/examples/faunadb-test.js)
-([EcmaScript 7 version](https://github.com/faunadb/faunadb-js/blob/master/examples/faunadb-test-es7.js)).
+[View reference JSDocs here](https://faunadb.github.com/faunadb-js).
 
-See the [FaunaDB Documentation](https://faunadb.com/documentation) for
-a complete API reference, or look in
-[`/tests`](https://github.com/faunadb/faunadb-python/tree/master/tests) for more
-examples.
+See the [FaunaDB Documentation](https://faunadb.com/documentation) for tutorials
+and a complete database API reference.
+
+## Supported Runtimes
+
+This Driver supports and is tested on:
+
+* Node.js
+  * LTS (v4)
+  * Stable (v6)
+  * v0.12.x
+* Chrome
+* Firefox
+* Safari
+* Internet Explorer 11
 
 ## Using the Client
 
-### Install
+### Installation
+
+#### Node.js
+
+TODO: NPM
+
+#### Browsers
 
-We will publish to `npm` soon.
+TODO: Browser source
 
 ### Use
 
-All work with FaunaDB happens through a Client instance.
-All Client methods return [Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).
-Although it's possible to use Client alone, it's often easier to do work through Model classes.
-For advanced work, the query api (`faunadb/lib/query`) will be useful.
+The [tutorials](http://faunadb.com/documentation/tutorials) in the FaunaDB documentation contain
+driver-specific examples.
+
+#### Requiring the Driver
+
+```javascript
+var faunadb = require('faunadb'),
+  q = faunadb.query,
+  Ref = q.Ref;
+  // Insert additional namespace aliases here.
+```
+
+This is the recommended require stanza. The `faunadb.query` module contains all of the
+functions to create FaunaDB Query expressions. This example also shows an example of
+defining a local `Ref` name, so as to not have to call `q.Ref` at all times.
+
+#### Instantiating a Client and Issuing Queries
+```javascript
+var client = new faunadb.Client({ secret: 'YOUR_FAUNADB_SECRET' });
+```
+
+Once the client has been instantiated, it can be used to issue queries. For example, to create
+an instance in an existing class named `test` with the data: `{ testField: 'testValue' }`:
 
+```javascript
+var createP = client.query(q.Create(Ref('classes/test'), { testField: 'testValue' }));
+```
 
-## Building it yourself
+Note that we are using the `Ref` alias defined above here.
 
-### Setup
+All methods on `faunadb.Client` return [ES6 Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).
+So, if we wanted to transform the Promise to return the `Ref` of the newly created instance:
 
-    npm install
+```javascript
+createP.then(function(res) {
+  return res.ref;
```

Comment:
I don't think a `return` is a good example of a promise continuation since the main scope won't have access to the desired ref until the callback function is executed.

I can see people doing things like:

```javascript
var ref  = createP.then(function(res) {
  return res.ref;
});

doSomething(ref); // failed because ref is null / undefined
```